ONSdigital/takeon-business-layer
# Feature/spp 1670 date adjustment

# Description
This is the update to the Date Adjustment method. We have added in the latest code from the smls-0.1.0b86 release. As a part of this,  modified business layer API's related to configuration, saving date adjustments into database and performed few schema changes by removing the average weekly column from Response table, merged 2 configuration table of date adjustment into single configuration table and also added a functionality to include expected start date and end dates when Q11 and Q12 responses are empty.

# How to test?
This is the update to the Date Adjustment method. We have added in the latest code from the smls-0.1.0b86 release. We have added unit tests in business layer to reflect the changes and added a check to see if the Date Adjustment method should be run for a particular survey

# Unit Tests
You can run java unit tests using: mvn clean test

# Code Coverage
You can verify code coverage using: awk -F"," '{ instructions += $4 + $5; covered += $5 } END { print covered, "/", instructions, " instructions covered"; print 100*covered/instructions, "% covered" }' target/site/jacoco/jacoco.csv

# Check Style errors
You can verify check style errors using : mvn package install

# Links
UI: https://dc-ui-spp1670.es-take-on.aws.onsdigital.uk/
JIRA: https://collaborate2.ons.gov.uk/jira/browse/SPP-1670
